app/scripts/lib/setupSentry.js

```diff
@@ -103,7 +103,17 @@ export default function setupSentry({ release, getState }) {
     environment,
     integrations: [new Dedupe(), new ExtraErrorData()],
     release,
-    beforeSend: (report) => rewriteReport(report),
+    beforeSend: (report) => {
+      if (getState) {
```

Comment:
I just tried testing this, and there is still a call made to Sentry upon `.init` being called. This effectively prevents error reports from being sent, but it doesn't delay all communication. We will need to come up with a different strategy.

Typically I would suggest delaying the call to `init`, but in our case that's challenging because _I suspect_ that `init` has side-effects that do not work correctly under lockdown.  It's possible they happen on import, not on `init`, but I haven't checked. If they happen on `init`, we're in a tough spot. It means that if we call it later, it'll fail.

Assuming that is true, we might be able to effectively stop all communication by passing in a custom transport. The transport option can be used to handle proxies and things like that, so I would expect it to affect absolutely all communication, even the initial call.

Apparently `init` doesn't support the `transport` option anymore despite it being documented, but it can be set by creating our own Sentry client manually. (details here: https://github.com/getsentry/sentry-javascript/issues/5422)

Another option would be to restart the process after opt-in, and somehow communicate to this bundle that the user opted in :thinking: Challenging since it would have to run before the controllers. This seems like a worse idea than the first one.

Reply:
I'm going to consider this non-blocking for this PR because what you have implemented so far is still an improvement, it's just not a complete solution yet.

Reply:
Okay, I will merge and then create a ticket to address the issues you have raised here.